AI/main_refactor.py

```
@@ -0,0 +1,842 @@
+def adminSendEmail():
+    if request.method == "POST":
+        emailType = request.form.get("emailType")
+        subject = request.form.get("subject")
+        message_body = request.form.get("message")
+
+        if emailType == "*":
+            emails = query_all("SELECT SchoolEmail FROM ACCOUNTS WHERE RoleID=0 AND (SchoolEmail IS NOT NULL)")
+            emails += query_all("SELECT HomeEmail FROM ACCOUNTS WHERE RoleID=0 AND (HomeEmail IS NOT NULL)")
+            emails = [e[0] for e in emails if e[0]]
+        else:
+            rows = query_all(f"SELECT {emailType} FROM ACCOUNTS WHERE RoleID=0 AND {emailType} IS NOT NULL")
```

Ruff (0.14.7)

Possible SQL injection vector through string-based query construction

(S608)